src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const currentState = { ...state };
+  const stateChangesArr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (true) {
```

Comment:
you can write switch (actions[i].type) and case 'addProperties', because expression in switch is checking on strict comparison to value in case